test/Ansible.common

```diff
@@ -9,8 +9,18 @@ $(ROLES_PATH) $(COLLECTIONS_PATH):
 
 .PHONY: ansible/dependencies
 ansible/dependencies: $(ROLES_PATH) $(COLLECTIONS_PATH)
+ifndef CROWDSTRIKE_ROLE_PULL_KEY
 	ansible-galaxy role install -r $(REQUIREMENTS_FILE) -p $(ROLES_PATH)
 	ansible-galaxy collection install -r $(REQUIREMENTS_FILE) -p $(COLLECTIONS_PATH)
+else ifeq ($(wildcard $(CROWDSTRIKE_ROLE_PULL_KEY)),)
+	ansible-galaxy role install -r $(REQUIREMENTS_FILE) -p $(ROLES_PATH)
+	ansible-galaxy collection install -r $(REQUIREMENTS_FILE) -p $(COLLECTIONS_PATH)
```

Comment:
Why are we doing the `ansible-galaxy` command before? Would it work to check if CROWDSTRIKE_ROLE_PULL_KEY is defined instead? If so, add the ssh key:

```
ifdef CROWDSTRIKE_ROLE_PULL_KEY
  eval $$(ssh-agent -s) && \
  ssh-add $(CROWDSTRIKE_ROLE_PULL_KEY)
endif
ansible-galaxy role install -r $(REQUIREMENTS_FILE) -p $(ROLES_PATH) && \
ansible-galaxy collection install -r $(REQUIREMENTS_FILE) -p $(COLLECTIONS_PATH)

Reply:
I wanted to do the `ssh-add` only if the variable is defined **and** the file exists in the host (if not the recipe might fail). Not sure Makefile syntax supported intersections so I wrote it like this. Is there any way we can avoid the duplication?

Reply:
I think the Makefile should fail (ssh-add will fail) if the file defined in CROWDSTRIKE_ROLE_PULL_KEY does not exist, instead of pulling the repos without it. I would rely on the caller to provide the expected file path. Indeed, what do you think if we call it `SSH_KEY` instead of CROWDSTRIKE_ROLE_PULL_KEY in the `Ansible.common` file, we can still keep the CROWDSTRIKE keyword in the provisioning.

Reply:
Not sure if we should fail if the CROWDSTRIKE_ROLE_PULL_KEY does not exist. I could be running this locally somehow, using other keys with permissions that the `ssh-agent` already has included, so I could pull the collections without this specific key. If no key is able to pull the private repo it will fail anyway, wdyt?

As for changing the var name, makes sense. Changed in https://github.com/newrelic/opentelemetry-collector-releases/pull/93/commits/291942074b2b22d7c49b96b926665e5f1886b65a!

Reply:
But wouldn't you have unexpected results? For the current example:
`$(MAKE) ansible/dependencies ANSIBLE_FOLDER=$(ANSIBLE_FOLDER) SSH_KEY=$(HOME)/.ssh/crowdstrike_ansible_role_key`

If `SSH_KEY=$(HOME)/.ssh/crowdstrike_ansible_role_key` does not exist, I won't be able to know I have provided an invalid path, but instead "some" roles will be fetched. I would stick with the fail-fast approach, but I might be missing something. We can discuss this async.